src/test/java/io/aiven/kafka/connect/s3/S3OutputStreamTest.java

```diff
@@ -16,63 +16,314 @@
 
-import com.amazonaws.auth.AWSStaticCredentialsProvider;
-import com.amazonaws.auth.BasicAWSCredentials;
-import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
 import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
-import io.findify.s3mock.S3Mock;
+import com.amazonaws.services.s3.model.AbortMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadResult;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadResult;
+import com.amazonaws.services.s3.model.PartETag;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+import com.amazonaws.services.s3.model.UploadPartResult;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
-public class S3OutputStreamTest {
+@ExtendWith(MockitoExtension.class)
+class S3OutputStreamTest {
+
+    static final String BUCKET_NAME = "some_bucket";
+
+    static final String FILE_KEY = "some_key";
+
+    static final String UPLOAD_ID = "some_upload_id";
+
+    @Mock
+    AmazonS3 mockedAmazonS3;
+
+    @Captor
+    ArgumentCaptor<InitiateMultipartUploadRequest> initiateMultipartUploadRequestCaptor;
+
+    @Captor
+    ArgumentCaptor<CompleteMultipartUploadRequest> completeMultipartUploadRequestCaptor;
+
+    @Captor
+    ArgumentCaptor<AbortMultipartUploadRequest> abortMultipartUploadRequestCaptor;
+
+    @Captor
+    ArgumentCaptor<UploadPartRequest> uploadPartRequestCaptor;
+
+    final Random random = new Random();
+
+    @Test
+    void noRequestsForEmptyBytes() throws Exception {
+
+        try (final var out = new S3OutputStream(BUCKET_NAME, FILE_KEY, 10, mockedAmazonS3)) {
+            out.write(new byte[] {});
+        }
+
+        verify(mockedAmazonS3, never()).initiateMultipartUpload(any(InitiateMultipartUploadRequest.class));
+        verify(mockedAmazonS3, never()).uploadPart(any(UploadPartRequest.class));
+        verify(mockedAmazonS3, never()).completeMultipartUpload(any(CompleteMultipartUploadRequest.class));
+        verify(mockedAmazonS3, never()).abortMultipartUpload(any(AbortMultipartUploadRequest.class));
```

Comment:
Maybe `verifyNoMoreInteractions(mockedAmazonS3);` instead of this 4 lines

Reply:
It checks that life-circle is ok and all methods involved into it.